pkg/internal/affinityassistant/affinityassistant_types_test.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2023 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package affinityassistant
+
+import (
+	"context"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	cfgtesting "github.com/tektoncd/pipeline/pkg/apis/config/testing"
+	"github.com/tektoncd/pipeline/test/diff"
+
+	"github.com/tektoncd/pipeline/pkg/apis/config"
+)
+
+func Test_GetAffinityAssistantBehavior(t *testing.T) {
+	tcs := []struct {
+		name      string
+		configMap map[string]string
+		expect    AffinityAssitantBehavior
+	}{{
+		name: "affinity-assistant-enabled",
+		configMap: map[string]string{
+			"disable-affinity-assistant": "false",
+		},
+		expect: AffinityAssistantPerWorkspace,
+	}, {
+		name: "affinity-assistant-disabled-coscheduling-workspaces",
+		configMap: map[string]string{
+			"disable-affinity-assistant": "true",
+			"coscheduling":               config.CoscheduleWorkspaces,
+		},
+		expect: AffinityAssistantDisabled,
+	}, {
+		name: "affinity-assistant-disabled-coscheduling-pipelineruns",
+		configMap: map[string]string{
+			"disable-affinity-assistant": "true",
+			"coscheduling":               config.CoschedulePipelineRuns,
+		},
+		expect: AffinityAssistantPerPipelineRun,
+	}, {
+		name: "affinity-assistant-disabled-coscheduling-isolate-pipelineruns",
+		configMap: map[string]string{
+			"disable-affinity-assistant": "true",
+			"coscheduling":               config.CoscheduleIsolatePipelineRuns,
+		},
+		expect: AffinityAssistantPerPipelineRunWithIsolation,
+	}, {
+		name: "affinity-assistant-disabled-coscheduling-disabled",
+		configMap: map[string]string{
+			"disable-affinity-assistant": "true",
+			"coscheduling":               config.CoscheduleDisabled,
+		},
+		expect: AffinityAssistantDisabled,
+	}}
+
+	for _, tc := range tcs {
+		ctx := cfgtesting.SetFeatureFlags(context.Background(), t, tc.configMap)
+		get, _ := GetAffinityAssistantBehavior(ctx)
```

Comment:
can you handle this error with t.Fatalf?

Reply:
Sure, added.